### ISSUE TITLE
CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected

### ISSUE THREAD

**Issue by ctueting** [deepmind/alphafold#381]
Hi all,

tl;dr 
GPUs were not recognized, so I reinstalled CentOS. 

I try to install AlphaFold on one of our workstations, but, as the title already says, the GPUs are not used.

The system is a workstation with 2 NVIDIA 2080 TIs and running centos:
```
$ cat /etc/centos-release
CentOS Linux release 7.9.2009 (Core)
```

Here is what I did so far:

I installed NCIDIA Container Toolkit and docker and set it up to run as a non-root user:
```
$ docker --version
Docker version 20.10.12, build e91ed57
```
And I can see the gpus:
```
$ docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi
Thu Feb 24 10:17:05 2022
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.54       Driver Version: 510.54       CUDA Version: 11.6     |
```
(output shortened)

Than I created a virtual environment for alphafold:
```
$ conda create -n alphafold
$ conda activate alphafold
$ conda install python=3.9.7
```

installed the requirements
`$ python3 -m pip install -r docker/requirements.txt`

and build the Docker image:
`$ docker build -f docker/Dockerfile -t alphafold .`

The databases were already downloaded and placed on /scratch/alphafold_db. I hardcoded this path in the run_docker.py file.

To test the alphafold installation, I downloaded a random sequence form Uniprot:
`$ wget https://uniprot.org/uniprot/P62837.fasta`

and run alphafold with this sequence
`$ python3 docker/run_docker.py --fasta_paths=P62837.fasta --max_template_date=2022-01-01 --model_preset=monomer_ptm`

and get the following issue:
```
I0224 11:35:36.788937 140230044059456 run_docker.py:247] 2022-02-24 10:35:36.788274: E external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
I0224 11:35:36.789192 140230044059456 run_docker.py:247] I0224 10:35:36.788722 140564657022784 xla_bridge.py:212] Unable to initialize backend 'gpu': Failed precondition: No visible GPU devices.
I0224 11:35:36.789365 140230044059456 run_docker.py:247] I0224 10:35:36.788864 140564657022784 xla_bridge.py:212] Unable to initialize backend 'tpu': Invalid argument: TpuPlatform is not available.
I0224 11:35:36.789520 140230044059456 run_docker.py:247] W0224 10:35:36.788920 140564657022784 xla_bridge.py:215] No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
```
(output reduced to the relevant part. Databases were mounted and JackHMMER started afterwards)

So it looks like tensorflow is not seeing the GPUs at all.

I remember, I had the same issue on another machine, but unfortunately I can't remember how I fixed it.

Thanks for your help.

**Comment by ctueting** [comment]
I somehow break the system while figuring out, what the issue. Other software, like cryoSPARC, failed also, trying to use the GPUs. So I ended up reinstalling CentOS, and just installed the minimal software needed for alphafold (conda, docker, nvidia-drivers) and it now runs.

The reason, everything fails might be the /dev/nvidia* devices. The group was neither root nor my user, so I tried to chown it back to root. This might cause eventually the error on all software.

Nevertheless, I cannot give any additional information on this case, to it might be closed.

Sorry for the inconvenience.
